rocky/rocky/views/organization_settings.py

```diff
@@ -1,9 +1,39 @@
+from datetime import datetime
+
+import time
+from enum import Enum
+from timeit import timeit
+
+from django.contrib import messages
+from django.core.exceptions import PermissionDenied
+from django.http import Http404, HttpRequest, HttpResponse
+from django.utils.translation import gettext_lazy as _
 from django.views.generic import TemplateView
 from django_otp.decorators import otp_required
 from tools.view_helpers import OrganizationDetailBreadcrumbsMixin
 from two_factor.views.utils import class_view_decorator
 
 
+class PageActions(Enum):
+    RECALCULATE = "recalculate"
+
+
 @class_view_decorator(otp_required)
 class OrganizationSettingsView(OrganizationDetailBreadcrumbsMixin, TemplateView):
     template_name = "organizations/organization_settings.html"
+
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        """Perform actions based on action type"""
+        action = request.POST.get("action")
+        if not self.request.user.has_perm("tools.recalculate_bits"):
+            raise PermissionDenied()
+        if action == PageActions.RECALCULATE.value:
+            connector = self.octopoes_api_connector
+
+            start_time = datetime.now()
+            number_of_bits = connector.recalculate_bits()
+            duration = datetime.now() - start_time
+            messages.add_message(request, messages.INFO, _(f"Recalculated {number_of_bits} bits. Duration: {duration}"))
+            return self.get(request, *args, **kwargs)
+        else:
```

Comment:
The `else` clause is redundant since if any of the other conditions above are met we will hit this raise statement anyway

Reply:
But it does make clear to the reader that we raise the Exception if action is something else than `PageActions.RECALCULATE.value`. But 404 is the wrong error code, because the organization settings page exists so it shouldn't give a 404. I think it would be more appropriate to return `HttpResponseBadRequest` or raise `SuspiciousOperation`